Rudim/Perft/PerftTest.cs

```diff
@@ -0,0 +1,58 @@
+﻿using CommandLine;
+using Rudim.Board;
+using Rudim.Common;
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Rudim.Test.Perft
+{
+
+    public class PerftTest
+    {
+        public void Perft(int depth, ulong nodes, string position)
+        {
+            var timer = new Stopwatch();
+
+            History.ClearBoardHistory();
+
+            timer.Start();
+
+            var boardState = BoardState.ParseFEN(position);
+            PerftDriver.ResetNodeCount();
+            PerftDriver.Traverse(boardState, depth);
+
+            timer.Stop();
+
+            if(nodes != PerftDriver.Nodes){
+                Console.WriteLine($"There's a difference Expected: {nodes} Actual: {PerftDriver.Nodes}");
+                Environment.Exit(2);
+            }
+
+            History.ClearBoardHistory();
+            Console.WriteLine($"Execution Time: {timer.ElapsedMilliseconds} ms");
```

Comment:
Can you add some data on which test it is - before with XUnit we would be able to know which perft is taking how much.

"Execution Time: 100ms for Starting Position at depth 5"

You can check ToString() for BoardState already has some logic for common names so you can probably just to boardState.ToString() to print the position